### ISSUE TITLE
Is this sort of a plugin on top of Theos?

### ISSUE THREAD

**Issue by HenkPoley** [leeghwrd/Development-Directory#1]
E.g.: https://github.com/theos/theos

It's doesn't quite say so in the README.md. Just that the .theos directory made me look it up.

Would it be in any way shape or form possible to have a terminal emulator where this lives inside that app, without jailbreaking? I mean there already are side-loadable apps that run a x86 emulator (lots of OPcode translation overhead) with iCloud Drive integration (Ex.: iSH).

**Comment by leeghwrd** [comment]
TL;DR, I made this package to get around the 'killed 9' error received when trying to run c/c++ programs I compiled on device. 

CoreTrust introduced in iOS 12 which is aimed at preventing the “fake signing” technique used by tools such as ldid and jtool. So trying to run my compiled programs from /var/moblie caused the killed 9 error.

After some investigation I realized that these same programs were able to run under /opt/development, That's where this package comes in. It just creates the /opt/development and adds a symbolic link to /var/mobile.

The ldid package has been receiving updates so this problem may be fixed now. I haven't tested.

But besides c/c++, it also allowed me to run NodeJS programs which I couldn't do from /var/mobile directory. Like I said i haven't tested in a while.

**Comment by HenkPoley** [comment]
Ah, so there is already a compiler [that can be installed when jailbroken] on the iPad ?

**Comment by leeghwrd** [comment]
Yup iPhone and iPad. I tried out some C++17 code yesterday and it compiled using an up to date toolchain.

**Comment by HenkPoley** [comment]
I'd rather not introduce security problems on an iOS installation. But if a jailbreak would make it possible to use an iPad as a full unix development environment, without relying on an external 'real' computer, that would be nice.

I remember it being janky back in the day. For example everything is wiped across iOS upgrades. You will need to run old insecure iOS (yes, [watering hole attacks exists](https://googleprojectzero.blogspot.com/2019/08/a-very-deep-dive-into-ios-exploit.html)). And the jailbreak community is not that security conscious overall.

I'm also looking at [UTM](https://github.com/utmapp/UTM), but it appears to do even full blown binary translation with ARM64-on-ARM64. I suspect a 20x performance impact. Which is sort of besides the point, any Raspberry Pi would be faster.

One can dream of a "iOS subsystem for {macOS, Darwin, Linux}" like WSL on Windows 10.

**Comment by HenkPoley** [comment]
Just for giggle I've jailbroken my old iPhone 5s, but I'm kind of stuck on how to tell some `./configure/` where the standard library lives.

`CC=/usr/bin/clang LDFLAGS="-l$THEOS/sdks/iPhoneOS11.2.sdk/" CPPFLAGS="-iSystem" ./configure --build=aarch64-apple-darwin18.7.0`

```
configure:4193: /usr/bin/clang  -iSystem -l/var/root/theos/sdks/iPhoneOS11.2.sdk/ conftest.c  >&5
clang-5.0: warning: argument unused during compilation: '-iSystem' [-Wunused-command-line-argument]
ld: library not found for -l/var/root/theos/sdks/iPhoneOS11.2.sdk/
Not signing file
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
```
